### PR TITLE
[chore] [exporter/awsxrayexporter] add metadata.yaml

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** Add metadata.yaml

**Link to tracking Issue:** #21213, #21479

### CODE REVIEW

**Comment by bryan-aguilar** [comment]
@atoulme do you have any info on these lint failures?
```
 factory_test.go:22: File is not `gci`-ed with --skip-generated -s standard,default,prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)
	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter/internal/metadata"
factory_test.go:29: File is not `gci`-ed with --skip-generated -s standard,default,prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)

config_test.go:21: File is not `gci`-ed with --skip-generated -s standard,default,prefix(github.com/open-telemetry/opentelemetry-collector-contrib) (gci)
	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter/internal/metadata"
```

**Review by atoulme — APPROVED** [review]

File: exporter/awsxrayexporter/doc.go [hunk 1]
```diff
@@ -14,4 +14,6 @@
 
 // Package awsxrayexporter implements an OpenTelemetry Collector exporter that sends trace data to
 // AWS X-Ray in the region the collector is running in using the PutTraceSegments API.
+
+//go:generate mdatagen metadata.yaml
```
Comment (codeboten) [inline exporter/awsxrayexporter/doc.go:18]: Nit: the godoc is usually left above the package definition and the go:generate line is above that doc string

**Review by codeboten — APPROVED** [review]

File: exporter/awsxrayexporter/doc.go (hunk 1 above)
Comment (bryan-aguilar) [reply exporter/awsxrayexporter/doc.go:18]: intellij auto formatting annoyingly disagrees. But I have forced it to do as i say!

Comment (codeboten) [reply exporter/awsxrayexporter/doc.go]: Weird! I wonder if there's a golint rule for this

**Pull request merged**